### DISCUSSION TITLE
Swagger Ui ? or Api  Documentation

### DISCUSSION THREAD

**Discussion by gunduzcagri** [lllyasviel/Fooocus#1765] [General]
Does it have Swagger UI feature? or any API?

If you know, can you write it with documentation?

**Comment by mashb1t** [comment]
Fooocus currently has a Gradio Websocket API, (see footer after running it).
If you want to use a REST API there is a fork which supports this: https://github.com/konieshadow/Fooocus-API
btw there already is a feature request foir ading HTTP REST API support: https://github.com/lllyasviel/Fooocus/issues/1224
Also see https://github.com/lllyasviel/Fooocus/issues/554, https://github.com/lllyasviel/Fooocus/issues/1120, https://github.com/lllyasviel/Fooocus/issues/1293 for examples.

Hope this helps!

**Reply by up_the_irons** [reply]
Is there any documentation on the Gradio Websocket API?  I searched for "websocket" on the sidebar at https://www.gradio.app/docs , but it came up with nothing 😭

**Reply by mashb1t** [reply]
Please find it in the footer of Fooocus. Websocket clients are available for JS and Python.
EDIT: When consulting the docs, you can find them in the bottom of the sidebar.

**Reply by up_the_irons** [reply]
Thanks!

**Reply by up_the_irons** [reply]
I installed Gradio and have a small working app:

```
const app = await client("http://0.0.0.0:7865/");
const result = await app.predict(24, []);
console.log(result.data);
```
This was from one of the examples in the sidebar.

A data structure was returned in `result.data`.

The problem is, all the documentation shows are unnamed routes.  No idea which route to use to generate an image.

Is there any documentation for this?  Should there be named routes, but I'm just not seeing them (for whatever reason)?

**Reply by mashb1t** [reply]
This is indeed correct, no route is named in Fooocus API (i assume due to backwards compatibility). You can temporarily add an api_name to a gradio action to identify the route.

**Reply by up_the_irons** [reply]
But I still don't know which route to use, even if I name them.  There are 40 or so unnamed routes.  Any idea how to differentiate between which ones do what action?

**Reply by toutjavascript** [reply]
@up_the_irons 
You must try and search.
And very often, function number changes...
It is tedious

**Reply by up_the_irons** [reply]
😭
